src/components/smart_objects/include/smart_objects/enum_schema_item.h

```diff
@@ -314,6 +317,21 @@ const ElementSignature TEnumSchemaItem<EnumType>::getSignature(
   return ret;
 }
 
+template <typename EnumType>
+bool TEnumSchemaItem<EnumType>::filterInvalidEnums(
+    SmartObject& Object,
+    const utils::SemanticVersion& MessageVersion,
+    rpc::ValidationReport* report__) {
+  rpc::ValidationReport dummy_report("");
+  if (validate(Object, &dummy_report, MessageVersion, false) != errors::OK) {
+    std::string validation_info =
+        "Filtered invalid value - " + Object.asString();
```

Comment:
Not sure if this is the validation info passed back to mobile but if it is, then it needs to follow the format mentioned in the proposal:

> Invalid enums were removed: <param_1>:<enum_value_1>,<enum_value_2>;<param_2>:<enum_value_3> ...